src/main/java/com/google/devtools/build/lib/buildtool/ExecutionTool.java

```diff
@@ -987,17 +987,17 @@ private Builder createBuilder(
   @VisibleForTesting
   public static void configureResourceManager(ResourceManager resourceMgr, BuildRequest request) {
     ExecutionOptions options = request.getOptions(ExecutionOptions.class);
-    ImmutableMap<String, Float> extraResources =
-        options.localExtraResources.stream()
+    ImmutableMap<String, Double> cpuRam =
+            ImmutableMap.of("cpu", options.localCpuResources, "memory", options.localRamResources);
+    ImmutableMap<String, Double> resources =
+       java.util.stream.Stream.concat(cpuRam.entrySet().stream(), options.localExtraResources.stream())
             .collect(
                 ImmutableMap.toImmutableMap(
                     Map.Entry::getKey, Map.Entry::getValue, (v1, v2) -> v2));
```

Comment:
IIUC, users will be able to override `--local_{cpu,ram}_resources` here because there isn't a check to ensure that users don't specify `--local_extra_resources={cpu,memory}=?`.  Because `localExtraResources.stream()` is concatenated after `cpuRam.stream()` and are collected into a single map here, the former overrides the latter. 

Since this map is now the source of truth (i.e. cpu and memory don't have separate fields in the `ResourceSet`), Bazel won't ignore any `"cpu"` or `"memory"` keys in the map as it previously did.

Lets keep the previous behavior of still having the `--local_{cpu,ram}_resources` flag (at least in this change). So you'd want to either flip the order of the concat, or filter out `"cpu"` and `"memory"` strings from `localExtraResources`.

Reply:
Updated. I was a bit on the fence on what was best when I first made this change, because ideally I'd be nice to have one attribute to deal with all resources (see proposal in https://github.com/bazelbuild/bazel/issues/19679). I do agree however that --local_**extra**_resources still carries that meaning of "resources other than some others". Might be worth adopting an incompatible flag to introduce `--local_resources`, WDYT? Happy to do it as a follow up as well.